File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.length();
        //ESWAR
        int max_length=0;
     for (int i = 0; i < n; i++) {
        int hash[256] = {0}; 
        int count=0;
        for (int j = i; j < n; j++) {
            if (hash[s[j]]) break; 
             hash[s[j]] = 1;         
             count++;                
        }
        max_length=max(count,max_length);
     }
        return max_length;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int max_length = 0;
        for (int i = 0; i < n; i++) {
            int count=0;
            int[] hash = new int[256]; 
            for (int j = i; j < n; j++) {
                if (hash[s.charAt(j)] == 1) break;
                hash[s.charAt(j)] = 1;
                count++;
            }
            max_length = Math.max(max_length, count);
        }
        return max_length;
    }
}